### DISCUSSION TITLE
How to keep current Layout with NuxtLink?

### DISCUSSION THREAD

**Discussion by robert-gruner** [nuxt/nuxt#25537] [Questions]
Hi @ll,

we are struggling a bit with using NuxtLink in combination with NuxtLayout. We do not have a "default" layout since there is no sensible default. We have a scenario where using NuxtLink leads to "forgetting" the currently set layout or maybe trying to set the default layout again (?). 

See prepared Stackblitz to showcase the problem: https://stackblitz.com/edit/github-wgg4gm-qrf88k

Repro:
1. Open app, see the initial layout
2. Go to products
3. Scroll down and try to go to other page via Pagination
4. For 1/2 we go to new route with whitescreen (no layout)
5. Page 3 button does nothing although I try to set new path

What are we doing wrong? Is there any chance to keep the currently applied layout when navigating via NuxtLink?

Thanks in advance for the help!

Regards,
Robert

**Comment by undermoonn** [comment]
There is nothing wrong :)

Route component switches when navigating. If you want to keep the current route component, you should use `keepAlive` and remove all layouts codes

<img width="197" alt="2024-02-03 23 14 21" src="https://github.com/nuxt/nuxt/assets/39266501/dd6736ab-8e0d-42cd-8c45-0b2f88f49d7f">

```vue
// products/[...page].vue
<script setup lang="ts">
import { storeToRefs } from 'pinia';

definePageMeta({
  keepalive: true,
  key() {
    return '/products'
  }
})

const route = useRoute()
const { products } = storeToRefs(useFakeStore()); // dont't forgot to request products with page index

watch(() => route.params.page, _page => {
  // fetch data when pagination
  // push or unshift data to products
})

</script>
```

Hope this can help you.

**Reply by robert-gruner** [reply]
Thanks for the idea. I'm afraid that is not what we need. This feature has to do with the Vue [KeepAlive](https://vuejs.org/guide/built-ins/keep-alive)  right?

We definitely cannot "remove all layouts codes" since the routing via URLs has no correlation to the necessary page layout in our case.

We are just wondering why NuxtLink or rather navigateTo resets the current layout to the "default". Our workaround for the moment is to save the currently applied layout and "reapply" it via routing middleware again.

**Reply by undermoonn** [reply]
> For 1/2 we go to new route with whitescreen (no layout)
<img width="259" alt="2024-02-06 15 51 37" src="https://github.com/nuxt/nuxt/assets/39266501/5b271729-44bd-489b-9199-cf6bf09830e3">

Do you want to keep products layout event if target path is not includes with `/products` 🤔

```ts
if (to.path.includes('/products')) {
   return setPageLayout('products')
}
```

**Reply by robert-gruner** [reply]
It is just a showcase, I could develop it even a bit further if you like. The 1/2/3 links should showcase a pagination feature, where pagination means we should stay on the same page type, same layout but route to a "paginated" subroute of the page. So the goal is that we keep the "products" layout but route to a products/s3.html for example.
